app/src/main/java/org/dhis2/utils/customviews/PeriodAdapter.kt

```diff
@@ -0,0 +1,80 @@
+package org.dhis2.utils.customviews
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.RecyclerView
+import java.util.ArrayList
+import java.util.Date
+import java.util.Locale
+import org.dhis2.R
+import org.dhis2.databinding.ItemDateBinding
+import org.dhis2.utils.DateUtils
+import org.hisp.dhis.android.core.period.PeriodType
+
+private class PeriodAdapter(
+    private val periodType: PeriodType,
+    openFuturePeriods: Int
+) : RecyclerView.Adapter<DateViewHolder>() {
+
+    companion object {
+        const val DEFAULT_PERIODS_SIZE = 10
+    }
+
+    private val datePeriods: MutableList<Date>
+    private var onDateSetListener: PeriodDialog.OnDateSet? = null
+    private var lastDate: Date
+
+    init {
```

Comment:
You could inline this init with the variables

Reply:
I believe that the way the it's done is the right way. According to kotlin's documentation _since the primary constructor cannot contain any code. Initialization code can be placed in initializer block_. but the declaration of the variables doesn't have to be inside.